walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/util/MeasureTime.kt

```diff
@@ -25,24 +26,39 @@ internal inline fun <R> logTime(name: String, block: () -> R): R {
     return result
 }
 
-internal inline fun <S> logNetworkTime(name: String, block: () -> Response<S>): Response<S> {
+internal inline fun logNetworkTime(name: String, block: () -> Result<IResponse>): Result<IResponse> {
     val start = System.currentTimeMillis()
-    val result = block()
+    var code = 0
+    var cvResponse = "none"
+    var requestId = "none"
+    val result = block().onSuccess {
+        cvResponse = it.headers[Constants.CORRELATION_VECTOR_HEADER] ?: "none"
+        requestId = it.headers[Constants.REQUEST_ID_HEADER] ?: "none"
+        code = it.status.toInt()
+    }.onFailure {
+        when(it) {
+            is IHttpAgent.ClientError -> {
```

Comment:
Does the Client and Server error behave differently here? Shouldn't this be a generic HttpAgentError where all of them collect the same information from the response?